### PR TITLE
Add missing TINS_API to PDU classes. Fix ICMPv6 enums

### PR DESCRIPTION
**Pull request by einarjon**
Extend/fix ICMPv6 enum values and unify naming

### CODE REVIEW

File: include/tins/pdu_cacher.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ namespace Tins {
  * etc.
  */
 template <typename T>
-class PDUCacher : public PDU {
+class TINS_API PDUCacher : public PDU {
```
Comment (mfontanini) [inline include/tins/pdu_cacher.h]: Is this needed for template classes? From the nature of templates, I would assume they shouldn't be imported by users of the library, as the full definition is always available.

File: include/tins/icmpv6.h [hunk 2]
```diff
@@ -80,14 +80,20 @@ class TINS_API ICMPv6 : public PDU {
         NEIGHBOUR_SOLICIT = 135,
         NEIGHBOUR_ADVERT = 136,
         REDIRECT = 137,
-        ROUTER_RENUMBER = 137,
+        ROUTER_RENUMBER = 138,
```
Comment (mfontanini) [inline include/tins/icmpv6.h]: Ouch, dem copy paste. Good catch

File: include/tins/pdu_cacher.h (hunk 1 above)
Comment (einarjon) [inline include/tins/pdu_cacher.h]: Probably not - I just used a "class.*public PDU" regexp...

On 17 February 2016 at 18:00, Matias Fontanini <notifications@github.com>
wrote:

> In include/tins/pdu_cacher.h
> <https://github.com/mfontanini/libtins/pull/129#discussion_r53195386>:
>
> > @@ -53,7 +53,7 @@ namespace Tins {
> >   * etc.
> >   */
> >  template <typename T>
> > -class PDUCacher : public PDU {
> > +class TINS_API PDUCacher : public PDU {
>
> Is this needed for template classes? From the nature of templates, I would
> assume they shouldn't be imported by users of the library, as the full
> definition is always available.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mfontanini/libtins/pull/129/files#r53195386>.
>



-- 
Regards
Einar Jón
+43 650 291 3472

Comment (mfontanini) [inline include/tins/pdu_cacher.h]: Can you remove it? I'll merge it after you do that.

Comment (einarjon) [inline include/tins/pdu_cacher.h]: I pushed it yesterday (commit f88d94c
<https://github.com/einarjon/libtins/commit/f88d94cbaa73d6eb4246fbcd2e7e173666253486>
).

It seems to be part of the pull request already, but this is my first
github contribution so I have no idea what I'm doing.

Thanks,
Einar
.

On 17 February 2016 at 18:52, Matias Fontanini <notifications@github.com>
wrote:

> In include/tins/pdu_cacher.h
> <https://github.com/mfontanini/libtins/pull/129#discussion_r53202904>:
>
> > @@ -53,7 +53,7 @@ namespace Tins {
> >   * etc.
> >   */
> >  template <typename T>
> > -class PDUCacher : public PDU {
> > +class TINS_API PDUCacher : public PDU {
>
> Can you remove it? I'll merge it after you do that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mfontanini/libtins/pull/129/files#r53202904>.
>



-- 
Regards
Einar Jón
+43 650 291 3472

**Comment by mfontanini** [comment]
Yeah, github doesn't notify me if you update the branch, so I didn't notice the commit, sorry.

Seems like you merge master into develop now, so now you're actually pushing my latest changes to master/develop again. If you run this commands to fix your branch I'll merge it right away (so you remove the last "Merge remote-tracking..." commit):

```Shell
# This is the "Remove TINS_API from PDUCacher" commit, so you'll erase the last one
git reset --hard f88d94cbaa73d6eb4246fbcd2e7e173666253486
git push origin master -f
```

To avoid having those "Merge ..." commits, instead of running `git pull upstream master`, you should run:

```Shell
git fetch upstream
git rebase upstream/master
```

This way, you apply your branch's commits after updating your local index to point to wherever upstream/master is pointing. This doesn't involve any commits at all, which is nice.

**Comment by mfontanini** [comment]
Nervemind, this doesn't affect anything. I'm merging it now.

**Pull request merged**